pkgs/development/tools/rust/cargo-flash/cargo-lock.patch

```diff
@@ -0,0 +1,35 @@
+diff --git a/Cargo.lock b/Cargo.lock
+index 6649de1..affa641 100644
+--- a/Cargo.lock
++++ b/Cargo.lock
+@@ -552,7 +552,9 @@ dependencies = [
+ 
+ [[package]]
+ name = "gdb-server"
+-version = "0.6.0"
++version = "0.6.1"
++source = "registry+https://github.com/rust-lang/crates.io-index"
++checksum = "c6e7b1809fa47255da438a2bddec1ec92e8d35628877c4dc24b44c86c3ba64f7"
+ dependencies = [
+  "async-std",
+  "futures",
+@@ -955,7 +957,9 @@ checksum = "b4596b6d070b27117e987119b4dac604f3c58cfb0b191112e24771b2faeac1a6"
+ 
+ [[package]]
+ name = "probe-rs"
+-version = "0.6.0"
++version = "0.6.2"
++source = "registry+https://github.com/rust-lang/crates.io-index"
++checksum = "9b4508248f016eb7f7a50751b9c3e8f9e25add169feac006aef5115dbb5e42be"
+ dependencies = [
+  "base64",
+  "bitfield",
```

Comment:
These dependency versions are [pinned precisely](https://github.com/probe-rs/cargo-flash/blob/4d60a470e1530ac8203f1248fbdcf4d1fd41e745/Cargo.toml#L24-L25) in the upstream `Cargo.toml`; if there's fixes from the new versions that are important to pick up, it would probably be a good idea to send a PR bumping the `Cargo.{toml,lock}` upstream.

Reply:
Indeed. I guess they do this on purpose, to keep the version in sync. However, the build breaks without this. What do we do here?

Reply:
Huh, that build error is really strange; for some reason cargo is insistent on updating probe-rs to a version that isn't what the Cargo.toml specifies, even though the versions specified aren't yanked or anything.

I think patching the lock file is reasonable for now, but bumping the versions evidently fixes things for users (us!) in practice, it'd be nice if the cargo-flash upstream could tag a new release with them.

Reply:
>  cargo is insistent on updating probe-rs to a version that isn't what the Cargo.toml specifies

A few days ago I heard about this behavior to be intentional, cargo updates dependencies sometimes even if passed the `--freeze` argument and a `Cargo.lock`. The [docs](https://doc.rust-lang.org/cargo/commands/cargo-install.html) for `--frozen` state this (which implicitly explains the observation):

> Either of these flags requires that the Cargo.lock file is up-to-date. If the lock file is missing, or it needs to be updated, Cargo will exit with an error.